pkg/kv/kvclient/kvcoord/dist_sender.go

```diff
@@ -1916,7 +1916,7 @@ func (ds *DistSender) sendToReplicas(
 			// evaluating twice, overwriting another unrelated write that fell
 			// in-between.
 			//
-			if withCommit && !grpcutil.RequestDidNotStart(err) {
+			if withCommit && !grpcutil.RequestDidNotStart(err) && !grpcutil.IsFailedPreconditionError(err) {
```

Comment:
shouldn't `RequestDidNotStart` be taught to recognize both AuthError and FailedPreconditionError as guaranteeing that the request didn't start?